src/track/globaltrackcache.cpp

```diff
@@ -464,25 +437,27 @@ TrackPointer GlobalTrackCache::lookupById(
 
 TrackPointer GlobalTrackCache::lookupByRef(
         const TrackRef& trackRef) {
-    TrackPointer trackPtr;
     if (trackRef.hasId()) {
-        trackPtr = lookupById(trackRef.getId());
+        TrackPointer trackPtr = lookupById(trackRef.getId());
         if (trackPtr) {
             return trackPtr;
         }
     }
     if (trackRef.hasCanonicalLocation()) {
-        trackPtr = lookupByCanonicalLocation(trackRef.getCanonicalLocation());
+        TrackPointer trackPtr = lookupByCanonicalLocation(trackRef.getCanonicalLocation());
         if (trackPtr) {
-            const auto cachedTrackRef =
-                    validateAndCanonicalizeRequestedTrackRef(trackRef, *trackPtr);
             // Multiple tracks may reference the same physical file on disk
-            if (!trackRef.hasId() || trackRef.getId() == cachedTrackRef.getId()) {
+            const auto cachedTrackRef = createTrackRef(*trackPtr);
+            if (trackRef.getLocation() == cachedTrackRef.getLocation()) {
                 return trackPtr;
             }
+            kLogger.warning()
+                    << "Found a different track for the same canonical location:"
+                    << "requested =" << trackRef
+                    << "cached =" << cachedTrackRef;
         }
     }
-    return trackPtr;
+    return {};
```

Comment:
```suggestion
    // Lookup failed, either because the TrackRef did not specify an id or canonical location,
    // or because the database contains multiple tracks with that canonical location.
    return {};
```